app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1098,6 +1098,10 @@ private boolean checkAndStartUnsentFormsTask(final boolean syncAfterwards,
     @Override
     protected void onResume() {
         super.onResume();
+        if (redirectedToLogin) {
+            finish();
+            return;
```

Comment:
@phillipm Need your opinion here- I realized that what we discussed earlier about having the `SessionAwareCommCareActivity` call `finish()` isn't sufficient, because the child activity will still proceed with whatever code path is in its `onResume()` method, even after `finish()` has been called. In order to truly halt all further execution of code in the activity, you have to call `finish()` AND `return`. But obviously, `return` has to be called from within the child activity's own `onResume` method... so I think in order to make this work, every single class that implements `SessionAwareCommCareActivity` would have to include the above lines of code in its `onResume` method.

So, 2 questions: Can you think of a better way of doing this that I'm not thinking of? And if not, do you think the mess of this is worth doing?